src/test/java/org/kainos/ea/service/ServiceTest.java

```diff
@@ -0,0 +1,94 @@
+package org.kainos.ea.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.kainos.ea.api.AuthService;
+import org.kainos.ea.cli.Login;
+import org.kainos.ea.cli.Register;
+import org.kainos.ea.cli.Role;
+import org.kainos.ea.client.FailedToGenerateTokenException;
+import org.kainos.ea.client.FailedToLoginException;
+import org.kainos.ea.client.FailedToRegisterException;
+import org.kainos.ea.core.Validator;
+import org.kainos.ea.db.AuthDao;
+
+import java.lang.reflect.Method;
+import java.sql.SQLException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class ServiceTest {
+
+    private AuthDao authDao;
+    private Validator validator;
+    private AuthService authService;
+
+    @BeforeEach
+    public void setUp() {
+        authDao = mock(AuthDao.class);
+        validator = mock(Validator.class);
+        authService = new AuthService(authDao, validator);
+    }
+
+    @Test
+    public void testLogin() throws FailedToLoginException, FailedToGenerateTokenException {
+        Login login = new Login("test@example.com", "password");
+        when(validator.isValidLogin(any(Login.class))).thenReturn(null);
+        when(authDao.getHashedPassword(anyString())).thenReturn("hashedPassword");
+        when(authDao.generateJwtToken(anyString())).thenReturn("token");
+
+        AuthService authServiceSpy = spy(authService);
+        doReturn(true).when(authServiceSpy).verifyPassword(anyString(), anyString());
+
+        String token = authServiceSpy.login(login);
+
+        assertEquals("token", token);
+        verify(authDao, times(1)).getHashedPassword(anyString());
+        verify(authDao, times(1)).generateJwtToken(anyString());
+    }
+
+
```

Comment:
only 1 unit test for each of login and registration, not testing any of the error secarios